include/onnxruntime/core/framework/allocator.h

```diff
@@ -73,51 +73,46 @@ struct OrtDevice {
 };
 
 struct OrtAllocatorInfo {
-  // use string for name, so we could have customized allocator in execution provider.
-  const char* name;
-  int id;
-  OrtMemType mem_type;
-  OrtAllocatorType type;
+  OrtAllocatorType allocator_type;
   OrtDevice device;
+  OrtMemType mem_type;
 
-  constexpr OrtAllocatorInfo(const char* name_, OrtAllocatorType type_, OrtDevice device_ = OrtDevice(), int id_ = 0, OrtMemType mem_type_ = OrtMemTypeDefault)
+  constexpr OrtAllocatorInfo(OrtAllocatorType allocator_type_, OrtDevice device_ = OrtDevice(), OrtMemType mem_type_ = OrtMemTypeDefault)
 #if (defined(__GNUC__) || defined(__clang__))
```

Comment:
Please help me remove 
```
#if (defined(__GNUC__) || defined(__clang__))
      __attribute__((nonnull))	      __attribute__((nonnull))
#endif	#endif
```
If there is no pointers in this function's parameter list.